internal/corazawaf/transaction.go

```diff
@@ -752,6 +752,51 @@ func (tx *Transaction) RequestBodyWriter() io.Writer {
 	return tx.RequestBodyBuffer
 }
 
+type Lenger interface {
```

Comment:
This is an attempt to check the Len before reading the Reader.

Reply:
I guess for proxy-wasm we will turn the bytes into a buffer.

Reply:
Let's have two functions, `WriteRequestBody(b []byte)` and `ReadRequestBodyFrom(r io.Reader)`. It's probably ok to wrap into buffer in this code but it should be worth it to not lose UX vs the current API.